Sources/SwiftFormat/Rules/AlwaysUseLiteralForEmptyCollectionInit.swift

```diff
@@ -0,0 +1,208 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftParser
+
+/// Never use `[<Type>]()` syntax. In call sites that should be replaced with `[]`,
+/// for initializations use explicit type combined with empty array literal `let _: [<Type>] = []`
+/// Static properties of a type that return that type should not include a reference to their type.
+///
+/// Lint:  Non-literal empty array initialization will yield a lint error.
+/// Format: All invalid use sites would be related with empty literal (with or without explicit type annotation).
+@_spi(Rules)
+public final class AlwaysUseLiteralForEmptyCollectionInit : SyntaxFormatRule {
```

Comment:
Do you think we need `Init` in the name here? Not that other rule names aren't already a mouthful, but it seems like we could drop it and have the same meaning.

Reply:
I think it’s useful to have it in the name to indicate that this is for initialization only, but I am open to suggestions if you have a better name in mind :)

Reply:
I can't think of anything better off the top of my head, so if you think it's helpful to disambiguate, I'm ok with it 🙂

Reply:
We can always change it later if something better comes up :)